core/controllers/topic_editor_test.py

```diff
@@ -145,6 +148,43 @@ def test_get(self):
                 ), expected_status_int=401)
             self.logout()
 
+    def test_get_questions_with_multi_skills(self):
+        for i in range(0, 4): #pylint: disable=unused-variable
+            question_id = question_services.get_new_question_id()
+            self.save_new_question(
+                question_id, self.admin_id,
+                self._create_valid_question_data('ABC'))
+            question_services.create_new_question_skill_link(
+                question_id, self.skill_id, 0.5)
+            question_services.create_new_question_skill_link(
+                question_id, self.skill_id_2, 0.3)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_EDITORS', True):
+            self.login(self.ADMIN_EMAIL)
+            with self.swap(constants, 'NUM_QUESTION_SKILL_LINKS_PER_PAGE', 6):
+                json_response = self.get_json(
+                    '%s/%s?cursor=' % (
+                        feconf.TOPIC_EDITOR_QUESTION_URL, self.topic_id
+                    ))
+                question_summary_dicts = json_response['question_summary_dicts']
+                self.assertEqual(len(question_summary_dicts), 3)
+                next_start_cursor = json_response['next_start_cursor']
+                json_response = self.get_json(
+                    '%s/%s?cursor=%s' % (
+                        feconf.TOPIC_EDITOR_QUESTION_URL, self.topic_id,
+                        next_start_cursor
+                    ))
+                question_summary_dicts_2 = (
+                    json_response['question_summary_dicts'])
+                self.assertEqual(len(question_summary_dicts_2), 1)
+                self.assertEqual(
```

Comment:
Ideally try to do a full assertEqual on question_summary_dicts and question_summary_dicts_2 to ensure that the contents are exactly what you expect. It feels like this is testing too little.

Reply:
Done.